### ISSUE TITLE
Uploading releases to S3 broken in 1.4.6

### ISSUE THREAD

**Issue by bencates** [edeliver/edeliver#263]
### Precheck

* Please do not use the issues tracker for help or support (try [the wiki](https://github.com/boldpoker/edeliver/wiki), [elixir forum](https://elixirforum.com/c/elixir-questions/deployment), [#deployment on slack](https://elixir-lang.slack.com/), Stack Overflow, IRC, mailing list, etc.)
* For proposing a new feature, please start a discussion at the [deployment category on the elixir forum](https://elixirforum.com/c/elixir-questions/deployment)
* For bugs, do a quick search and make sure the bug has not yet been reported
* Finally, be nice and have fun!

### Environment

* Edeliver version (`mix edeliver --version`): 1.4.6
* Elixir version (`elixir -v`): 1.5.2
* Build tool and version (distillery): distillery
* Operating system (on build / deploy hosts): Ubuntu 14.04.5
* Are you using an umbrella project (yes|no): no

### Verbose output

```
> mix edeliver build release

BUILDING RELEASE OF HAL APP ON BUILD HOST

-----> Authorizing hosts
-----> Ensuring hosts are ready to accept git pushes
-----> Pushing new commits with git to: ubuntu@buildserver
-----> Resetting remote hosts to bad18b7aa0a39a0366bf6470561f78c3cc3dec33
-----> Cleaning generated files from last build
-----> Fetching / Updating dependencies
-----> Compiling sources
-----> Generating release
-----> Copying release 2.3.0+406-bad18b7 to s3 release store
bash: line 3497: warning: here-document at line 4 delimited by end-of-file (wanted `EOF#!/usr/bin/perl')
syntax error at - line 3493, near "; >"
Unknown regexp modifier "/n" at - line 3493, at end of line
Regexp modifiers "/u" and "/l" are mutually exclusive at - line 3493, at end of line
Regexp modifiers "/u" and "/l" are mutually exclusive at - line 3493, at end of line
Execution of - aborted due to compilation errors.

A remote command failed on:

  ubuntu@52.37.170.44

Output of the command is shown above if the --verbose option
was used and the commands executed on that remote host are
printed below for debugging purposes:

FAILED with exit status 255:

ssh -o ConnectTimeout=3 ubuntu@buildserver 

      [ -f ~/.profile ] && source ~/.profile
      set -e
      cd /var/builds/hal/_build/prod/rel/hal/releases/2.3.0+406-bad18b7 > /dev/null
      AWS_ARGUMENTS="put system76-builds/hal/hal_2.3.0+406-bad18b7.release.tar.gz hal.tar.gz" AWS_ACCESS_KEY_ID="##REDACTED##" \
      AWS_SECRET_ACCESS_KEY="##REDACTED##" perl > /dev/null <<'EOF'
        content from libexec/aws file
      EOF ; > /dev/null
```

The issue disappears when the --verbose flag is passed. I'm pretty sure the issue is [this change](https://github.com/edeliver/edeliver/commit/ec1129b16e81de137a842b10953afd76a830dc76#diff-12b95fb389714e84ff22577bee1abb27L483).

### Current behavior

Building a release with an S3 release store successfully builds the release but errors out when trying to upload it to S3.

### Expected behavior

Building a release with an S3 release store copies the release to S3.

**Comment by lessless** [comment]
Hi @bencates ,

Thanks for looking into issue. According to shellcheck https://github.com/koalaman/shellcheck/wiki/SC1121 this is indeed a misformed instructions. Can you please try it with:

```
 AWS_SECRET_ACCESS_KEY="##REDACTED##" perl > /dev/null <<'EOF'
content from libexec/aws file
EOF
```

**Comment by bencates** [comment]
Yes. Running it with `--verbose` (which sets `$SILENCE` to empty string and therefore produces the snippet above) works as expected.

**Comment by lessless** [comment]
I assume that problem is in the  `EOF ; > /dev/null` which was on the last line of the snippet.

Here is where it supposedly happens https://github.com/edeliver/edeliver/blob/master/libexec/erlang#L312 

```
local _aws_script_content=$(cat $BASE_PATH/libexec/aws)
#...
perl $SILENCE <<'EOF'${_aws_script_content}EOF ;" "$HOSTS_APP_USER" "
```
The `perl $SILENCE <<'EOF'${_aws_script_content}EOF ;` is fine as we know, but then a weird thing happens - an unescaped quotes. I will take a look on how `aws` command works and will try to reproduce it.

**Comment by lessless** [comment]
@bencates after a quick `aws` script inspection I didn't saw any need for the additional arguments and thus assume that `$HOSTS_APP_USER` is safe to delete, which I did in https://github.com/edeliver/edeliver/tree/try-to-fix-263 branch.

Can you please try it out? Notice, that it's **not tested**  !

**Comment by lessless** [comment]
@syamilmj It's definitely a `$SILENCE` variable being appended to the operation, but I don't see where it's coming from so I just reverted changes to it introduced in the #147 
Can you please try `revert-silence-redirection-changes` branch?

**Comment by lessless** [comment]
Sorry, didn't have an intention to close an issue in the commit. Let's ensure that problem is solved before doing that.

**Comment by syamilmj** [comment]
@lessless Can confirm that `revert-silence-redirection-changes` works for me.

**Comment by ryanwinchester** [comment]
@lessless is there something you'd reccommend I can do currently to have this working?

- Use master branch?
- Revert to an older version?
- ...?

**Comment by lessless** [comment]
Hi @ryanwinchester !
I just rolled 1.5 with latest fixes. Try it out!

**Comment by ryanwinchester** [comment]
@lessless 

```
{:edeliver, github: "edeliver/edeliver", tag: "1.5.0"}
```

works for me, thanks.

**Comment by lessless** [comment]
Sorry, forgot to publish on hex. It's there now ;)
